dbms/src/Encryption/RateLimiter.h

```diff
@@ -46,6 +47,37 @@ enum class LimiterType
     BG_READ = 4,
 };
 
+/// IOInfo is used to store IO information.
+/// total_write_bytes is the total bytes of the write.
+/// total_read_bytes is the total bytes of the read.
+/// bg_write_bytes is the bytes of the background write.
+/// bg_read_bytes is the bytes of the background read.
+/// update_time is the time of the last update.
+struct IOInfo
```

Comment:
Currently, what we need from `IOInfo` is `bg_read_bytes` and `fg_read_bytes = total_read_bytes - bg_read_bytes`. I think  use atomic variable to handle concurrent access of these two variables is a better than `std::shared_mutex`.